plugins/modules/aws_api_gateway_domain.py

```diff
@@ -0,0 +1,333 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: aws_api_gateway_domain
+short_description: Manage AWS API Gateway custom domains
+description:
+     - Manages API Gateway custom domains for API GW Rest APIs.
+     - AWS API Gateway custom domain setups use CloudFront behind the scenes.
+       So you will get a CloudFront distribution as a result, configured to be aliased with your domain.
+version_added: '3.2.0'
```

Comment:
One last change request @stefanhorning 🙈  

```suggestion
version_added: '3.3.0'
```

Reply:
sure. that was unavoidable :)